### ISSUE TITLE
Post-process Time

### ISSUE THREAD

**Issue by refdinal22** [ultralytics/ultralytics#22731]
Why is the post-processing time in the latest release significantly higher compared to version 8.3.163?

**Pretrained model (YOLO11x.pt)**

- Latest Release (8.3.229)

![Image](https://github.com/user-attachments/assets/6c40cac0-871a-44ab-8749-91c157b97c7a)

- 8.3.163

![Image](https://github.com/user-attachments/assets/bb0208fa-c421-42ad-88fa-a528d77a5e24)

**Custom Dataset**

- Latest Release (8.3.229)

![Image](https://github.com/user-attachments/assets/999fcc2c-5262-48ae-86d3-bbe673d8fb2e)

- 8.3.163

![Image](https://github.com/user-attachments/assets/68ff3813-db37-468b-bef5-3e8a44db7f6a)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @refdinal22, thanks for reporting this and for sharing detailed comparisons across versions and models 🚀 This is an automated initial response to help gather key details; an Ultralytics engineer will follow up with you soon for deeper assistance.

Since your issue is about performance differences (post-processing time) between versions, it’s very helpful for us to see exactly how you are running your benchmarks.

If this is a 🐛 Bug Report related to regression or performance changes, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) that clearly shows:
- The exact code snippet used for inference and timing
- The `ultralytics` versions tested (e.g. 8.3.163 vs 8.3.229)
- Hardware details (GPU/CPU, OS)
- Model used (e.g. `YOLO11x.pt` or your custom model path)
- Any custom settings (NMS args, image size, batch size, device, etc.)

This will help us determine whether the increased post-process time is due to library changes, environment differences, or configuration changes 💡

If this is a custom training or usage ❓ Question about optimizing or interpreting the timing breakdown, please also include:
- How the model was trained (for your custom dataset)
- Example inference logs or screenshots (like the ones you attached)
- Confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)

In the meantime, we recommend reviewing the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that may help you double-check your benchmarking setup.

You’re also welcome to discuss performance topics with the community:
- For real-time chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth threads, see [Discourse](https://community.ultralytics.com/)
- For community insights and shared benchmarks, visit our [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before we dig deeper, please make sure you’ve tested with the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), to confirm the behavior persists:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

If possible, try to reproduce the timing differences in one of these environments and include the details in your reply ✅

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you share an MRE and environment details, it will greatly speed up the investigation into the post-processing time change ⚙️

**Comment by Y-T-G** [comment]
You can add

```
import torchvision
```
in your code

**Comment by refdinal22** [comment]
Alright, It’s already solved.
Thank you

**Comment by glenn-jocher** [comment]
Glad it’s sorted out, and thanks for confirming—if you run into anything else with post-processing or YOLO11 performance, please open a new issue with details.
